subprojects/ide-native/src/integTest/groovy/org/gradle/ide/xcode/XcodeSwiftExternalSourceDependenciesIntegrationTest.groovy

```diff
@@ -0,0 +1,360 @@
+            version = '2.0'
+
+            dependencies {
+                implementation "org.test:greeter:latest.integration"
+            }
+        """
+        fixture.executable.writeToProject(testDirectory)
+
+        when:
+        succeeds 'xcode'
+
+        then:
+        executedAndNotSkipped(":greeter:compileDebugSwift", ":xcodeProject", ":xcodeProjectWorkspaceSettings", ":xcodeSchemeAppExecutable", ":xcode")
+        rootXcodeWorkspace.contentFile.assertHasProjects("${rootProjectName}.xcodeproj")//, "${checkoutRelativeDir(repo.name, commit.id.name, repo.id)}/greeter.xcodeproj")
+
+        def appProject = xcodeProject("${rootProjectName}.xcodeproj").projectFile
+        appProject.indexTarget.getBuildSettings().SWIFT_INCLUDE_PATHS == toSpaceSeparatedList(checkoutDir(repo.name, commit.id.name, repo.id).file('build/modules/main/debug'))
+    }
+
+    def "adds source dependencies Swift module of test component to Xcode indexer search path"() {
+        def library = new SwiftLib()
+        def test = new SwiftLibTest(library, library.greeter, library.sum, library.multiply)
+
+        given:
+        buildTestFixture.withBuildInSubDir()
+        singleProjectBuild("greeter") {
+            buildFile << """
+                apply plugin: 'swift-library'
+            """
+            library.writeToProject(it)
+        }
+        def commit = repo.commit('initial commit', GFileUtils.listFiles(file('greeter'), null, true))
+
+
+        and:
+        settingsFile << """
+            sourceControl {
+                vcsMappings {
+                    withModule("org.test:greeter") {
+                        from vcs(GitVersionControlSpec) {
+                            url = "${repo.url}"
+                        }
+                    }
+                }
+            }
+        """
+        buildFile << """
+            apply plugin: 'swift-library'
+            apply plugin: 'xcode'
+            apply plugin: 'xctest'
+
+            dependencies {
+                testImplementation "org.test:greeter:latest.integration"
+            }
+        """
+        test.writeToProject(testDirectory)
+
+        when:
+        succeeds 'xcode'
+
+        then:
+        executedAndNotSkipped(":greeter:compileDebugSwift", ":xcodeProject", ":xcodeProjectWorkspaceSettings", ":xcodeSchemeAppSharedLibrary", ":xcode")
+        rootXcodeWorkspace.contentFile.assertHasProjects("${rootProjectName}.xcodeproj")
+
+        def appProject = xcodeProject("${rootProjectName}.xcodeproj").projectFile
+        appProject.targets.find { it.isUnitTest() }.getBuildSettings().SWIFT_INCLUDE_PATHS == toSpaceSeparatedList(file('build/modules/main/debug'), checkoutDir(repo.name, commit.id.name, repo.id).file('build/modules/main/debug'))
+        appProject.indexTarget.getBuildSettings().SWIFT_INCLUDE_PATHS == toSpaceSeparatedList(file('build/modules/main/debug'), checkoutDir(repo.name, commit.id.name, repo.id).file('build/modules/main/debug'))
```

Comment:
Why do we need to add the main module file here when we don't in the previous cases? I would guess that Xcode doesn't need that connection between the tests and production sources for a project to be made explicit like this. If we do add it to the include path, we should probably build it during generation, so that the result doesn't change when Gradle builds the modules later on.

This isn't related to this PR, but it's something we should tidy up as a separate issue.

Reply:
You are so right. My bad for not questioning this enough when developing this code. I will open an issue for this.

Reply:
Issue created: https://github.com/gradle/gradle-native/issues/331.